code/game/objects/items/cards_ids.dm

```diff
@@ -137,23 +144,75 @@
 		emagging = TRUE
 		if(do_after(user, rand(5, 10) SECONDS, target))
 			charges--
-			if (prob(40))
+			if(prob(40))
 				to_chat(user, span_notice("[src] emits a puff of smoke, but nothing happens."))
 				emagging = FALSE
 				return
-			if (prob(5))
+			if(prob(5))
 				var/mob/living/M = user
 				M.adjust_fire_stacks(1)
 				M.ignite_mob()
 				to_chat(user, span_danger("The card shorts out and catches fire in your hands!"))
 			log_combat(user, target, "attempted to emag")
-			if(istype(target, /obj/machinery/computer/bounty)) //we can't have nice things
-				to_chat(user, span_notice("The cheap circuitry isn't strong enough to subvert this!"))
-				emagging = FALSE
-				return
-			target.emag_act(user)
+			if(!target.emag_act(user, src) && !((charges + 1) > max_charges))
+				charges++
 		emagging = FALSE
 
+/obj/item/card/emag/cmag
+	desc = "It's a card coated in a slurry of electromagnetic bananium."
+	name = "jestographic sequencer"
+	icon_state = "cmag"
+
+/obj/item/card/emag/cmag/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/slippery, 8 SECONDS, GALOSHES_DONT_HELP) // It wouldn't be funny if it couldn't slip!
+
+#define CMAG_AIRLOCK_CLOSED	1 // Since this is undefined outside of `airlock.dm`, have to redefine them.
+#define CMAG_AIRLOCK_EMAG	6
+
+/obj/item/card/emag/cmag/afterattack(atom/target, mob/user, proximity)
+	. = ..() // Handles the ..() before that.
+	if(!proximity && prox_check)
+		return 
+	if(charges < 1)
+		to_chat(user, span_danger("\The [src] is still recharging!"))
+		return
+
+	log_combat(user, target, "attempted to cmag")
+	// Since cmag only has very few interactions, all of it is handled in `afterattack` instead of `emag_act`.
+	if(istype(target, /obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/airlock = target
+		if(airlock.operating || !airlock.density || !airlock.hasPower() || (airlock.obj_flags & EMAGGED) || (airlock.obj_flags & CMAGGED))
+			return
+
+		charges--
+		airlock.operating = TRUE
+		airlock.update_icon(state = CMAG_AIRLOCK_EMAG, override = TRUE)
+
+		sleep(0.6 SECONDS)
+		if(QDELETED(src))
+			return
+		airlock.update_icon(state = CMAG_AIRLOCK_CLOSED, override = TRUE)
+		airlock.obj_flags |= CMAGGED
+		airlock.operating = FALSE
```

Comment:
This is the same animation as emag, do we want this? It isn't technically opening anything, and it would be easier as a stealth thing that just adds the flag.

Reply:
Likely not, but I'd still like a way for someone to notice when someone is actively cmagging everything. So, I decided to replace it with a bikehorn sound.